McCorrections/src/PileUpWeight.cpp

```diff
@@ -115,16 +123,23 @@ double PileUpWeightEx::Get(EventInfo& eventInfo) const
 }
 double PileUpWeightEx::Get(const ntuple::ExpressEvent& event) const { return Get(event.npu); }
 
-double PileUpWeightEx::Get(double nPU) const
+double PileUpWeightEx::Get(const ntuple::ExpressEvent& event, UncertaintyScale unc_scale) const { return Get(event.npu, unc_scale); }
+
+double PileUpWeightEx::Get(double nPU, UncertaintyScale unc_scale) const
 {
     if(!active_group.is_initialized())
          throw exception("active group isn't initialized");
-    auto w_hist = pu_weights.at(*active_group);
+    auto w_hist = pu_weights_map.at(unc_scale).at(*active_group);
     const Int_t bin = w_hist->FindBin(nPU);
     const Int_t maxBin = w_hist->FindBin(max_available_pu);
     const bool goodBin = bin >= 1 && bin <= maxBin;
     return goodBin ? w_hist->GetBinContent(bin) : default_pu_weight;
 }
 
+double PileUpWeightEx::Get(EventInfo& eventInfo, UncertaintyScale unc_scale) const
+{
+    return Get(eventInfo->npu);
```

Comment:
why unc scale is not used? Should be the other way around Get without `unc_scale` calls Get with `unc_scale` setting it to its default value (`Central`)...